RecoParticleFlow/PFProducer/src/MLPFModel.cc

```diff
@@ -174,10 +234,28 @@ namespace reco::mlpf {
 
     math::PtEtaPhiELorentzVectorD p4(pred_pt, pred_eta, pred_phi, pred_e);
 
-    reco::PFCandidate cand(
-        0, math::XYZTLorentzVector(p4.X(), p4.Y(), p4.Z(), p4.E()), reco::PFCandidate::ParticleType(0));
-    cand.setPdgId(pred_pid);
-    cand.setCharge(charge);
+    reco::PFCandidate::ParticleType particleType(reco::PFCandidate::X);
+    if (pred_pid == 211)
```

Comment:
It looks like this `pred_pid` is unsigned (here and everywhere else in this code), i.e. it does not consider the charge: can you confirm, just for check<ì

Reply:
Yes, by construction, the underlying model reconstructs the absolute value of the PID separately from the charge.

So far, we didn't present detailed results about the charge prediction (so `//cand.setCharge(charge);` downstream), but it should not be a major problem, as should be driven by the track information.